llvm/lib/CodeGen/MachineBlockPlacement.cpp

```diff
@@ -1592,6 +1603,11 @@ bool MachineBlockPlacement::hasBetterLayoutPredecessor(
   BlockFrequency CandidateEdgeFreq = MBFI->getBlockFreq(BB) * RealSuccProb;
   bool BadCFGConflict = false;
 
+  // Compile-time optimization: runtime is quadratic in the number of
```

Comment:
this can probably be checked earlier in the function, to avoid extra work?

Reply:
Done